std/array.asm

```diff
@@ -17,7 +17,4 @@ let fold = |arr, initial, folder| std::utils::fold(len(arr), |i| arr[i], initial
 let sum: int[] -> int = |arr| fold(arr, 0, |a, b| a + b);
 
 /// Zips two arrays
-let zip2 = |array1, array2, fn| new(len(array1), |i| fn(array1[i], array2[i]));
-
-/// Zip three arrays
-let zip3 = |array1, array2, array3, fn| new(len(array1), |i| fn(array1[i], array2[i], array3[i]));
\ No newline at end of file
+let zip = |array1, array2, fn| new(len(array1), |i| fn(array1[i], array2[i]));
```

Comment:
Can you add a todo that we should assert equal length as soon as we have statements as part of expressions?

Reply:
I'll do in the next PR, as it's already in the queue